core/src/main/resources/xsl/raxRoles.xsl

```diff
@@ -113,15 +113,24 @@
 
     <xsl:template name="generateRoles">
         <xsl:param name="roles" as="xsd:string*" select="()"/>
-        <xsl:if test="not('#all' = $roles)">
-            <xsl:for-each select="$roles">
-                <wadl:param name="X-ROLES" style="header" rax:code="403" rax:anyMatch="true"
-                            rax:message="You are forbidden to perform the operation" type="xsd:string"
-                            required="true" repeating="true">
-                    <xsl:attribute name="fixed" select="rax:transformNBSP(.)"/>
-                </wadl:param>
-            </xsl:for-each>
-        </xsl:if>
+        <xsl:choose>
+            <xsl:when test="not('#all' = $roles)">
+                <rax:captureHeader name="X-RELEVANT-ROLES">
+                    <xsl:attribute name="path" select="string-join(('for $h in req:headers(''X-ROLES'', true()) return if ($h = (''', string-join($roles, ''', '''), ''')) then $h else ()'), '')"/>
```

Comment:
Man, I totally remember reading about attribute value templates in the past, but I don't touch XPath very much, and so did not remember them at all.

Anyway, I'll split up the processing a bit and use an attribute value template.